cli_tools/google-osconfig-agent/main.go

```diff
@@ -60,46 +90,33 @@ func main() {
 	}
 
 	logger.Init(ctx, proj)
+	defer logger.Close()
 
-	action := flag.Arg(0)
-	if action == "inventory" {
+	switch action := flag.Arg(0); action {
+	case "":
+		if err := service.Register(ctx, "google_osconfig_agent", "Google OSConfig Agent", "", run, "run"); err != nil {
```

Comment:
where did you previously run service.Register?

Reply:
In service.go, that separation and indirection was not accomplishing anything so I moved it here. 'register' here takes an action 'run' to register the service and run it, really this just runs the code on linux and on windows runs the code with the correct hooks for the windows service system. Othe args of install, uninstall, start in the service library can no longer get passed on the command line as the package installs do that work.